Apps/Src/SendTritium.c

```diff
@@ -0,0 +1,356 @@
+        }
+        // Sets brakelight
+        Minion_Write_Output(BRAKELIGHT, brakelightState, &minion_err);
+    }
+}
+
+// State Handlers & Deciders
+
+/**
+ * Normal Drive State. Accelerator is mapped directly to current setpoint.
+*/
+void NormalDriveHandler(){
+    if(reverseSwitch && !forwardSwitch){
+        velocitySetpoint = -MAX_VELOCITY;
+        currentSetpoint = percentToFloat(accelPedalPercent);
+    }else if(!reverseSwitch && forwardSwitch){
+        velocitySetpoint = MAX_VELOCITY;
+        currentSetpoint = percentToFloat(accelPedalPercent);
+    }
+    else if(!reverseSwitch && !forwardSwitch){
+        currentSetpoint = 0;
+    }
+    brakelightState = false;
+}
+
+void NormalDriveDecider(){
+    if(cruiseSet && cruiseEnable && brakePedalPercent == 0){
+        state = RECORD_VELOCITY;
+    }else if(!cruiseEnable && regenToggle && ChargeEnable){
+        state = ONEPEDAL_DRIVE;
+    }else if(brakePedalPercent > 0){
+        state = BRAKE_STATE;
+    }
+}
+
+/**
+ * Record Velocity State. While pressing the cruise set button, the car will record the observed velocity
+ * into velocitySetpoint.
+*/
+void RecordVelocityHandler(){
+    velocitySetpoint = MotorController_ReadVelocity();
+    brakelightState = true;
+}
+
+void RecordVelocityDecider(){
+    if(cruiseEnable && !cruiseSet){
+        state = POWERED_CRUISE;
+    }else if(!cruiseEnable){
+        state = NORMAL_DRIVE;
+    }else if(brakePedalPercent > 0){
+        state = BRAKE_STATE;
+    }
+}
+
+/**
+ * Powered Cruise State. Continue to travel at the recorded velocity as long as
+ * Observed Velocity <= Velocity Setpoint
+*/
+void PoweredCruiseHandler(){
+    currentSetpoint = 1.0f;
+    brakelightState = false;
+}
+
+void PoweredCruiseDecider(){
+    if(MotorController_ReadVelocity() > velocitySetpoint){
+        state = COASTING_CRUISE;
+    }else if(!cruiseEnable){
+        state = NORMAL_DRIVE;
+    }else if(!cruiseEnable && !cruiseSet){
+        state = RECORD_VELOCITY;
+    }else if(brakePedalPercent > 0){
+        state = BRAKE_STATE;
+    }
+}
+
+/**
+ * Coasting Cruise State. We do not want to utilize motor braking in cruise control mode.
+ * Coast the motor (do not motor brake) if we want to slow down; instead, stop putting power
+ * into the motor and let the car slow down naturally.
+*/
+void CoastingCruiseHandler(){
+    currentSetpoint = 0;
+    brakelightState = false;
+}
+
+void CoastingCruiseDecider(){
+    if(MotorController_ReadVelocity() <= velocitySetpoint){
+        state = POWERED_CRUISE;
+    }else if(!cruiseEnable){
+        state = NORMAL_DRIVE;
+    }else if(brakePedalPercent > 0){
+        state = BRAKE_STATE;
+    }
+}
+
+/**
+ * Brake State. Brake the car if the brake pedal is pressed.
+*/
+void BrakeHandler(){
+    currentSetpoint = 0;
+    velocitySetpoint = 0;
+    cruiseEnable = 0;
+    cruiseSet = 0;
+    brakelightState = true;
+}
+
+void BrakeDecider(){
+    if(brakePedalPercent == 0){
```

Comment:
Might want to replace this with checking if the brake pedal is below a certain threshold: I'm not sure what kind of noise we're going to see on the pedal ADCs this time around. It would be super inconvenient to go into the brake state for no reason all of the time.